src/opentracing-c/tracer.h

```diff
@@ -0,0 +1,160 @@
+     * must be zero.
+     */
+    int num_references;
+
+    /**
+     * Array of tags. May be NULL.
+     */
+    const opentracing_tag* tags;
+
+    /**
+     * Number of tags in array. If tags is NULL, num_tags must be zero.
+     */
+    int num_tags;
+} opentracing_start_span_options;
+
+/**
+ * Tracer interface.
+ * @extends opentracing_destructible
+ */
+typedef struct opentracing_tracer {
+    /** Base class member. */
+    opentracing_destructible base;
+
+    /**
+     * Close the tracer. This may do nothing, flush pending spans, or whatever
+     * else a tracing implementation decides close should mean. If not called
+     * before the destruction of the tracer, the tracer must perform the
+     * equivalent of close upon destruction.
+     * @attention The main distinction between close and destroy is whether or
+     *            not the tracer is in a valid state after the call.
+     *            Upon destruction, subsequent use of a tracer may result in
+     *            undefined behavior (i.e. members may be set to NULL causing
+     *            deference of NULL member). However, after close, the tracer
+     *            must not remain in an undefined state.
+     *
+     * @param tracer Tracer instance.
+     */
+    void (*close)(struct opentracing_tracer* tracer) OPENTRACINGC_NONNULL_ALL;
+
+    /**
+     * Equivalent to calling start_span_with_options with options as NULL.
+     * @param tracer Tracer instance.
+     * @param operation_name Name of operation associated with span.
+     * @return Span pointer on success, NULL otherwise.
+     */
+    opentracing_span* (*start_span)(struct opentracing_tracer* tracer,
+                                    const char* operation_name)
+        OPENTRACINGC_NONNULL_ALL;
+
+    /**
+     * Start a new span with provided options.
+     * @param tracer Tracer instance.
+     * @param operation_name Name of operation associated with span.
+     * @param options Options to override default span initialization values.
+     * @return Span pointer on success, NULL otherwise.
+     */
+    opentracing_span* (*start_span_with_options)(
+        struct opentracing_tracer* tracer,
+        const char* operation_name,
+        const opentracing_start_span_options* options) OPENTRACINGC_NONNULL(1,
+                                                                            2);
+
+    /**
+     * Inject span context into carrier.
+     * @param tracer Tracer instance.
+     * @param format Propagation format.
+     * @param carrier Opaque carrier.
+     * @param span_context Span context to serialize to carrier.
+     * @return Error code indicating success or failure.
+     */
+    opentracing_propagation_error_code (*inject)(
+        struct opentracing_tracer* tracer,
+        opentracing_propagation_format format,
+        void* carrier,
+        const opentracing_span_context* span_context) OPENTRACINGC_NONNULL_ALL;
```

Comment:
There was some discussion on https://github.com/opentracing/opentracing-go/issues/127 on getting rid of the format parameter, which is why it's not in the C++ API. Does it make sense to have it for the C API?